tracer/src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -123,14 +127,23 @@ private int WriteTags(ref byte[] bytes, int offset, Span span, ITagProcessor[] t
         {
             int originalOffset = offset;
 
-            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _metaBytes);
+            // Start of "meta" dictionary. Do not add any string tags before this line.
+            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, MetaBytes);
 
             int count = 0;
 
             // We don't know the final count yet, write a fixed-size header and note the offset
             var countOffset = offset;
             offset += MessagePackBinary.WriteMapHeaderForceMap32Block(ref bytes, offset, 0);
 
+            // write "language=dotnet" tag unless this span represents a call to an external service
+            if (this is not InstrumentationTags { SpanKind: SpanKinds.Client })
```

Comment:
Should we be adding it to Producer and Consumer spans then too? 🤔 It seems a little strange for Producer in particular, as those are "producing" to an external service?

Reply:
Makes sense. I excluded `Producer` spans too in 2b5dd44b632101506b362194e1982e6c344c6869.

`Consumer` spans are weird because even though they are "calls to an external dependency", they tend to be entry points into a service (root spans), so we probably want the tag there.

Reply:
> so we probably want the tag there

I agree, I think Consumer spans are like "server" spans